lumigator/python/mzai/backend/backend/services/datasets.py

```diff
@@ -75,12 +75,13 @@ def validate_experiment_dataset(filename: str):
 
 
 def dataset_has_gt(filename: str) -> bool:
-    with Path(filename).open() as f:
-        reader = csv.DictReader(f)
-        fields = set(reader.fieldnames or [])
-        has_gt = GT_FIELD in fields
+    dataset = load_dataset("csv", data_files=filename, split="train")
+
+    if GT_FIELD not in dataset.column_names:
+        return False
 
-    return has_gt
+    # True only if every value in dataset[GT_FIELD] is not None or empty.
+    return all(value for value in dataset[GT_FIELD])
```

Comment:
This treats the empty string of length `1` or more (i.e., this `" "` not this `""`) as `True`. I'm not against that, just recognizing that an entry of `" "` is considered valid ground truth. Are we okay with that?

Reply:
👍🏼 good point. I think this also leads to us wanting to have more unit test cases that cover the scenarios where we say something has ground truth or not.